src/motion/cp_lbfgs.F

```diff
@@ -269,12 +277,22 @@ SUBROUTINE setulb(n, m, x, lower_bound, upper_bound, nbd, f, g, factr, pgtol, wa
          END DO
       ENDIF
 
-      CALL mainlb(n, m, x, lower_bound, upper_bound, nbd, f, g, factr, pgtol, &
-                  wa(lws), wa(lwy), wa(lsy), wa(lss), wa(lwt), &
-                  wa(lwn), wa(lsnd), wa(lz), wa(lr), wa(ld), wa(lt), wa(lxp), &
-                  wa(lwa), &
-                  iwa(1), iwa(n + 1), iwa(2*n + 1), task, iprint, &
-                  csave, lsave, isave(22), dsave)
+      ! passes spgr to mainlb
+      IF (PRESENT(spgr)) THEN
```

Comment:
Same here. Just keep the `ELSE` block and pass the optional argument `spgr` forward.